entity_gym/entity_gym/environment.py

```diff
@@ -174,57 +183,136 @@ def clear(self) -> None:
 ActionMaskBatch = Union[CategoricalActionMaskBatch, SelectEntityActionMaskBatch]
 
 
+@dataclass
+class Entity:
+    features: List[str]
+
+
+@dataclass
+class ObsSpace:
+    entities: Dict[str, Entity]
+
+
 @dataclass
 class ObsBatch:
     entities: Dict[str, RaggedBufferF32]
-    ids: Sequence[Sequence[EntityID]]
-    action_masks: Mapping[str, ActionMaskBatch]
+    ids: List[Sequence[EntityID]]
+    action_masks: Dict[str, ActionMaskBatch]
     reward: npt.NDArray[np.float32]
     done: npt.NDArray[np.bool_]
     end_of_episode_info: Dict[int, EpisodeStats]
 
 
-def batch_obs(obs: List[Observation]) -> ObsBatch:
+def merge_obs(a: ObsBatch, b: ObsBatch) -> ObsBatch:
+    """
+    merges two obs_batches
+    """
+
+    entities = a.entities.copy()
```

Comment:
This creates a shallow copy, so we still end up mutating the original list. Easiest approach could be to just make the method mutate the first arg, if we make this a method on `ObsBatch` that returns `None` it would be fairly obvious that that's what's happening.